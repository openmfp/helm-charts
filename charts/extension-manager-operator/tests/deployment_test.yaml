suite: operator
templates:
  - deployment.yaml
  - service-account.yaml
  - cluster-role.yaml
  - cluster-rolebinding.yaml
values:
  - ../test-values.yaml
chart:
  version: 1.0.0
  appVersion: 1.0.0
release:
  name: extension-manager-operator
tests:
  - it: operator match the snapshot
    asserts:
      - matchSnapshot: {}
    set:
      istio:
        enabledOverride: true
      health:
        portOverride: 8081
        liveness:
          pathOverride: /readyz
      metrics:
        port: 8080
      deployment:
        resources:
          limits:
            cpuOverride: 260m
            memoryOverride: 512Mi
          requests:
            cpuOverride: 40m
            memoryOverride: 50Mi
  - it: deployment with metrics probes
    template: deployment.yaml
    set:
      health:
        port: 8081
        liveness:
          path: /healthz
      metrics:
        port: 8082
    asserts:
      - equal:
          path: spec.template.spec.containers[0].livenessProbe
          value:
            failureThreshold: 1
            httpGet:
              path: /healthz
              port: 8081
            periodSeconds: 10
  - it: check deployment template labels
    template: deployment.yaml
    set:
      deployment:
        template:
          labels:
            control-plane: controller-manager
    asserts:
      - equal:
          path: spec.template.metadata.labels.control-plane
          value: controller-manager
  - it: check deployment without custom labels
    template: deployment.yaml
    set:
      deployment:
        template: {}
    asserts:
      - equal:
          path: spec.template.metadata.labels
          value:
            app: extension-manager-operator
            service: extension-manager-operator
  - it: test imagePullPolicy override
    template: deployment.yaml
    set:
      imagePullPolicyOverride: IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: override imagePullSecret
    template: service-account.yaml
    set:
      imagePullSecretOverride: my-secret
    asserts:
      - equal:
          path: imagePullSecrets
          value: 
            - name: my-secret
  - it: configure global imagePullSecret
    template: service-account.yaml
    set:
      global:
        imagePullSecret: my-secret
    asserts:
      - equal:
          path: imagePullSecrets
          value: 
            - name: my-secret
  - it: with validationServer
    template: deployment.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[1].name
          value: server