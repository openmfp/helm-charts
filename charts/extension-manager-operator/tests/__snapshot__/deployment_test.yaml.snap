operator match the snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: extension-manager-operator
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - contentconfigurations
          - contentconfigurations/status
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - get
          - list
          - watch
          - create
          - update
          - patch
          - delete
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: extension-manager-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: extension-manager-operator
    subjects:
      - kind: ServiceAccount
        name: extension-manager-operator
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: extension-manager-operator
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: extension-manager-operator
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            service: extension-manager-operator
        spec:
          containers:
            - args:
                - operator
                - --leader-elect
                - --log-level=
                - --health-probe-bind-address=:8081
              automountServiceAccountToken: true
              image: ghcr.io/openmfp/extension-manager-operator:1.0.0
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
              name: manager
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              serviceAccountName: extension-manager-operator
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8081
                periodSeconds: 10
          serviceAccountName: extension-manager-operator
          terminationGracePeriodSeconds: 10
  4: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: extension-manager-operator
